src/extension.ts

```diff
@@ -210,29 +224,82 @@ export async function activate(context: vscode.ExtensionContext) {
 	);
 
 	context.subscriptions.push(
-		vscode.commands.registerCommand('intuita.createPR', async () => {
-			const panelInstance = IntuitaPanel.getInstance(
-				context,
-				{ getConfiguration },
-				globalStateAccountStorage,
-				messageBus,
-			);
+		vscode.commands.registerCommand('intuita.createPR', async (arg0) => {
+			try {
+				if (!repositoryService) {
+					throw new Error('Unable to initialize repositoryService');
+				}
 
-			//@TODO connect in next subtask
-			const baseBranch = sourceControl.getBaseBranchName();
-			const title = 'Title';
-			const body = 'Body';
-			const targetBranch = 'targetBranchName';
+				const currentBranch =
+					await repositoryService.getCurrentBranch();
 
-			await panelInstance.render();
-			panelInstance.setView({
-				viewId: 'createPR',
-				viewProps: {
-					initialFormData: { title, body, baseBranch, targetBranch },
-					loading: false,
-					error: '',
-				},
-			});
+				if (!currentBranch) {
+					throw new Error('Unable to get HEAD');
+				}
+
+				const hasChanges =
+					await repositoryService.hasWorkingTreeChanges();
+
+				if (!hasChanges) {
+					throw new Error('Nothing to commit');
+				}
+
+				const jobHash = typeof arg0 === 'string' ? arg0 : null;
+
+				if (jobHash === null) {
+					throw new Error(
+						`Could not decode the first positional arguments: it should have been a string`,
+					);
+				}
+
+				const treeItem = await treeDataProvider.getTreeItem(
+					jobHash as ElementHash,
+				);
+
+				const { label } = treeItem;
+				const jobTitle =
+					typeof label === 'object' ? label.label : label ?? '';
+
+				const panelInstance = IntuitaPanel.getInstance(
+					context,
+					{ getConfiguration },
+					globalStateAccountStorage,
+					messageBus,
+				);
+
+				// @TODO figure out more informative title and description
+				const title = jobTitle;
+				const body = 'Add description';
+
+				const targetBranch = repositoryService.getBranchName(
+					jobHash,
+					jobTitle,
+				);
+
+				await panelInstance.render();
+
+				// @TODO check why branch name can be undefined
+				const currentBranchName = currentBranch.name ?? '';
```

Comment:
A short explaination is when you're on (HEADed) a tag or the commit (detached HEAD)